// survey-angular-ui/esm2015/components/progress/toc/progress.component.js
import { Component, Input } from "@angular/core";
import { EmbeddedViewContentComponent } from "../../../embedded-view-content.component";
import { createTOCListModel, getTocRootCss } from "survey-core";
import { AngularComponentFactory } from "../../../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "../../list/list.component";
export class ProgressTocComponent extends EmbeddedViewContentComponent {
    constructor() {
        super(...arguments);
        this.containerCss = "sv_progress-toc";
    }
    createProgressTOCModel() {
        this.containerCss = getTocRootCss(this.model);
        this.listModel = createTOCListModel(this.model);
    }
    ngOnInit() {
        super.ngOnInit();
        this.createProgressTOCModel();
    }
    ngOnChanges(changes) {
        this.createProgressTOCModel();
    }
}
ProgressTocComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ProgressTocComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
ProgressTocComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ProgressTocComponent, selector: "sv-progress-toc, sv-ng-progress-toc", inputs: { model: "model" }, usesInheritance: true, usesOnChanges: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"containerCss\">\n    <sv-ng-list [model]=\"listModel\"></sv-ng-list>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.ListComponent, selector: "sv-ng-list, '[sv-ng-list]'", inputs: ["model"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ProgressTocComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-progress-toc, sv-ng-progress-toc",
                    templateUrl: "./progress.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("sv-progress-toc", ProgressTocComponent);
