// survey-angular-ui/esm2015/questions/composite.component.js
import { Component } from "@angular/core";
import { AngularComponentFactory } from "../component-factory";
import { QuestionAngular } from "../question";
import * as i0 from "@angular/core";
import * as i1 from "../panel.component";
export class CompositeQuestionComponent extends QuestionAngular {
    get contentPanel() {
        return this.model.contentPanel;
    }
}
CompositeQuestionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CompositeQuestionComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
CompositeQuestionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: CompositeQuestionComponent, selector: "sv-ng-composite-question", usesInheritance: true, ngImport: i0, template: "<sv-ng-panel [model]='contentPanel'></sv-ng-panel>", isInline: true, components: [{ type: i1.PanelComponent, selector: "sv-ng-panel, '[sv-ng-panel]'", inputs: ["model"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CompositeQuestionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-composite-question",
                    template: "<sv-ng-panel [model]='contentPanel'></sv-ng-panel>"
                }]
        }] });
AngularComponentFactory.Instance.registerComponent("composite-question", CompositeQuestionComponent);
