// survey-angular-ui/esm2015/questions/custom.component.js
import { Component } from "@angular/core";
import { QuestionAngular, getComponentName } from "../question";
import { AngularComponentFactory } from "../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "../utils/dynamic.directive";
export class CustomQuestionComponent extends QuestionAngular {
    get contentQuestion() {
        return this.model.contentQuestion;
    }
    getComponentName(element) { return getComponentName(element); }
}
CustomQuestionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CustomQuestionComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
CustomQuestionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: CustomQuestionComponent, selector: "sv-ng-custom-question", usesInheritance: true, ngImport: i0, template: "<ng-template [component]='{ name: getComponentName(contentQuestion), data: { model: contentQuestion } }'></ng-template>", isInline: true, directives: [{ type: i1.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CustomQuestionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-custom-question",
                    template: "<ng-template [component]='{ name: getComponentName(contentQuestion), data: { model: contentQuestion } }'></ng-template>"
                }]
        }] });
AngularComponentFactory.Instance.registerComponent("custom-question", CustomQuestionComponent);
