// survey-angular-ui/esm2015/questions/html.component.js
import { Component } from "@angular/core";
import { QuestionAngular } from "../question";
import { AngularComponentFactory } from "../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "../utils/safe-html.pipe";
export class HtmlQuestionComponent extends QuestionAngular {
    onModelChanged() {
        super.onModelChanged();
        this.model.locHtml.onChanged = () => {
            this.detectChanges();
        };
    }
    ngOnDestroy() {
        this.model.locHtml.onChanged = () => { };
        super.ngOnDestroy();
    }
}
HtmlQuestionComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: HtmlQuestionComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
HtmlQuestionComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: HtmlQuestionComponent, selector: "sv-ng-html-question", usesInheritance: true, ngImport: i0, template: "<div [class]=\"model.renderCssRoot\" [innerHTML]=\"model.locHtml.renderedHtml | safeHtml\" #contentElement></div>", pipes: { "safeHtml": i1.SafeHtmlPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: HtmlQuestionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-html-question",
                    templateUrl: "./html.component.html",
                    styleUrls: []
                }]
        }] });
AngularComponentFactory.Instance.registerComponent("html-question", HtmlQuestionComponent);
