// survey-angular-ui/esm2015/questions/multipletext.component.js
import { QuestionAngular } from "../question";
import { Component } from "@angular/core";
import { AngularComponentFactory } from "../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "./multipletextitem.component";
import * as i2 from "@angular/common";
export class MultipleTextComponent extends QuestionAngular {
    constructor() {
        super(...arguments);
        this.trackRowBy = (index) => {
            return this.model.inputId + "rowkey" + index;
        };
    }
    trackItemBy(_, item) {
        return "item" + item.editor.id;
    }
}
MultipleTextComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: MultipleTextComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
MultipleTextComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: MultipleTextComponent, selector: "sv-ng-multipletext-question", usesInheritance: true, ngImport: i0, template: "<table [class]=\"model.cssClasses.root\" #contentElement>\n  <tbody>\n    <tr\n      *ngFor=\"let row of model.getRows(); index as rowIndex; trackBy: trackRowBy\"\n      [class]=\"model.cssClasses.row\"\n    >\n      <ng-container *ngFor=\"let item of row; trackBy: trackItemBy\" >\n        <td [class]=\"model.cssClasses.cell\" [question]=\"model\" [model]=\"item\" sv-ng-multipletext-item></td>\n      </ng-container>\n    </tr>\n  </tbody>\n</table>\n", components: [{ type: i1.MultipleTextItemComponent, selector: "'[sv-ng-multipletext-item]'", inputs: ["question", "model"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: MultipleTextComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-multipletext-question",
                    templateUrl: "./multipletext.component.html"
                }]
        }] });
AngularComponentFactory.Instance.registerComponent("multipletext-question", MultipleTextComponent);
