// survey-angular-ui/esm2015/components/survey-header/logo-image.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory } from "../../component-factory";
import { EmbeddedViewContentComponent } from "../../embedded-view-content.component";
import * as i0 from "@angular/core";
import * as i1 from "../../utils/safe-url.pipe";
export class LogoImageComponent extends EmbeddedViewContentComponent {
    get survey() {
        return this.data;
    }
}
LogoImageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: LogoImageComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
LogoImageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: LogoImageComponent, selector: "sv-logo-image", inputs: { data: "data" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <img\n  [class]=\"survey.css.logoImage\"\n  [attr.src]=\"survey.locLogo.renderedHtml | safeUrl\"\n  [attr.width]=\"survey.logoWidth ? survey.logoWidth : undefined\"\n  [attr.height]=\"survey.logoHeight ? survey.logoHeight : undefined\"\n  [attr.alt]=\"survey.locTitle.renderedHtml\"\n  [style.objectFit]=\"survey.logoFit\"\n  />\n</ng-template>", styles: [":host{display:none}\n"], pipes: { "safeUrl": i1.SafeUrlPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: LogoImageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-logo-image",
                    templateUrl: "./logo-image.component.html",
                    styleUrls: ["../../hide-host.scss"]
                }]
        }], propDecorators: { data: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("sv-logo-image", LogoImageComponent);
