// survey-angular-ui/esm2015/components/character-counter/character-counter.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory } from "../../component-factory";
import { BaseAngular } from "../../base-angular";
import * as i0 from "@angular/core";
export class CharacterCounterComponent extends BaseAngular {
    getModel() {
        return this.counter;
    }
}
CharacterCounterComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CharacterCounterComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
CharacterCounterComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: CharacterCounterComponent, selector: "sv-ng-character-counter", inputs: { counter: "counter", remainingCharacterCounter: "remainingCharacterCounter" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"remainingCharacterCounter\">{{ counter.remainingCharacterCounter }}</div>\n</ng-template>", styles: [":host { display: none; }"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: CharacterCounterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-character-counter",
                    templateUrl: "./character-counter.component.html",
                    styles: [":host { display: none; }"],
                }]
        }], propDecorators: { counter: [{
                type: Input
            }], remainingCharacterCounter: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("character-counter", CharacterCounterComponent);
