// survey-angular-ui/esm2015/utils/ng-show.directive.js
import { Directive, Input } from "@angular/core";
import * as i0 from "@angular/core";
export class VisibleDirective {
    constructor(el) {
        this.el = el;
    }
    ngOnChanges(changes) {
        changes["visible"].currentValue ? this.show() : this.hide();
    }
    hide() {
        this.el.nativeElement.style.display = "none";
    }
    show() {
        this.el.nativeElement.style.display = "";
    }
}
VisibleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: VisibleDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
VisibleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.17", type: VisibleDirective, selector: "[visible]", inputs: { visible: "visible" }, usesOnChanges: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: VisibleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: "[visible]"
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { visible: [{
                type: Input
            }] } });
