// survey-angular-ui/esm2020/questions/customwidget.component.mjs
import { Component, Input, ViewChild } from "@angular/core";
import { AngularComponentFactory } from "../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../utils/dynamic.directive";
import * as i3 from "../utils/safe-html.pipe";
export class CustomWidgetComponent {
    get hasDefaultRender() {
        return this.model.customWidget.isDefaultRender || this.hasAngularComponent;
    }
    get hasHtml() {
        return this.model.customWidget.htmlTemplate ? true : false;
    }
    get customHtml() {
        return this.model.customWidget.htmlTemplate;
    }
    get hasAngularComponent() {
        return AngularComponentFactory.Instance.isComponentRegistered(this.model.customWidget.name);
    }
    get componentName() {
        if (this.hasAngularComponent)
            return this.model.customWidget.name;
        return this.model.getTemplate() + "-question";
    }
    ngAfterViewInit() {
        this.model.customWidget.afterRender(this.model, this.container.nativeElement);
    }
    ngOnChanges(simpleChanges) {
        if (simpleChanges["model"].previousValue !== undefined && simpleChanges["model"].currentValue !== undefined) {
            this.model.customWidget.afterRender(this.model, this.container.nativeElement);
        }
    }
    ngOnDestroy() {
        this.model.customWidget.willUnmount(this.model, this.container.nativeElement);
    }
}
CustomWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CustomWidgetComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
CustomWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.0.3", type: CustomWidgetComponent, selector: "sv-ng-custom-widget", inputs: { css: "css", model: "model" }, viewQueries: [{ propertyName: "container", first: true, predicate: ["content"], descendants: true }], usesOnChanges: true, ngImport: i0, template: "<div #content>\n  <div *ngIf=\"hasHtml\" [innerHTML]=\"customHtml | safeHtml\"></div>\n  <ng-container *ngIf=\"hasDefaultRender\">\n    <ng-template [component]=\"{ name: componentName, data: { model: model, css: css } }\"></ng-template>  \n  </ng-container>\n</div>", directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }], pipes: { "safeHtml": i3.SafeHtmlPipe } });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: CustomWidgetComponent, decorators: [{
            type: Component,
            args: [{ selector: "sv-ng-custom-widget", template: "<div #content>\n  <div *ngIf=\"hasHtml\" [innerHTML]=\"customHtml | safeHtml\"></div>\n  <ng-container *ngIf=\"hasDefaultRender\">\n    <ng-template [component]=\"{ name: componentName, data: { model: model, css: css } }\"></ng-template>  \n  </ng-container>\n</div>" }]
        }], propDecorators: { css: [{
                type: Input
            }], model: [{
                type: Input
            }], container: [{
                type: ViewChild,
                args: ["content"]
            }] } });
AngularComponentFactory.Instance.registerComponent("survey-customwidget", CustomWidgetComponent);
